File: isPrime.py
def IsPrime(number: int) -> bool:
    return False if number < 2 or number % 2 == 0 and number > 2 else len(list(filter(lambda denominator: number % denominator == 0, list(range(1, int(number/2)))))) < 2

def isPrime(number):
    if number < 2 or isinstance(number, float) :
        return False

    if number % 2 == 0:
        if number > 2:
            return False
        return True

    denominator = 1
    divisors = []
    while denominator < (number+1)/2:
        if number % denominator == 0:
            divisors.append(denominator)
        if len(divisors) == 2:
            return False
        denominator += 1

    return True


def main():
    #testing functions
    primes1 = list(filter(IsPrime, list(range(-10, 1000))))
    primes2 = list(filter(isPrime, list(range(-10, 1000))))
    print(primes1 == primes2, primes1, primes2)

if __name__ == "__main__":
    main()

File: isPrime.hs
{-
    Test if a giver number is a Prime or not.
    I even do not know if this work, but at least I tried.
-}
isPrime :: Integer -> Bool
isPrime number = "The number:" ++ show number ++ "is a Prime number? " ++ show result ++ "."
    where result = if number < 2 || number `mod` 2 == 0 && number > 2 then False else length (filter (\denominator -> number `mod` denominator == 0) [2..number/2]) < 2


{-
    This one was made by ChatGPT
-}
IsPrime :: Int -> Bool
IsPrime number
  | number < 2 || (number `mod` 2 == 0 && number > 2) = False
  | otherwise = length (filter (\denominator -> number `mod` denominator == 0) [1..(number `div` 2)]) < 2
